sktime/performance_metrics/forecasting/_classes.py

```diff
@@ -68,391 +75,463 @@ class _BaseForecastingErrorMetric(BaseMetric):
+    }
 
+    # todo: 0.13.0, remove the greater_is_better property
+    @property
+    def greater_is_better(self):
+        """Whether `fit` has been called."""
+        warn(
+            "The greater_is_better attribute is deprecated from 0.12.0 "
+            "and will be removed in 0.13.0. Use the lower_is_better tag instead, "
+            'e.g., my_metric.get_tag("lower_is_better")',
+            DeprecationWarning,
+        )
+        return not self.get_tag("lower_is_better", False)
+
+    def __init__(
+        self,
+        func=None,
+        name=None,
+        multioutput="uniform_average",
+        multilevel="uniform_average",
+    ):
+        self.multioutput = multioutput
+        self.multilevel = multilevel
+        # todo: in 0.13.0, when func/name are removed, remove func/name args
+        #   temporarily, this is how we avoid overwrite with None
+        # super().__init__(func=func, name=name)
+        self.name = type(self).__name__
 
-class _PercentageErrorMixin:
     def __call__(self, y_true, y_pred, **kwargs):
         """Calculate metric value using underlying metric function.
 
-        Uses `symmetric` attribute to determine whether underlying function
-        should return symmetric percentage error metric or a percentage error
-        metric.
-
         Parameters
         ----------
-        y_true : pd.Series, pd.DataFrame or np.array of shape (fh,) or \
-                (fh, n_outputs) where fh is the forecasting horizon
-            Ground truth (correct) target values.
-
-        y_pred : pd.Series, pd.DataFrame or np.array of shape (fh,) or  \
-                (fh, n_outputs)  where fh is the forecasting horizon
-            Forecasted values.
-
-        y_train : pd.Series, pd.DataFrame or np.array of shape (n_timepoints,) or \
-                (n_timepoints, n_outputs), default = None
-            Optional keyword argument to pass training data.
-
-        y_pred_benchmark : pd.Series, pd.DataFrame or np.array of shape (fh,) \
-             or (fh, n_outputs) where fh is the forecasting horizon
-            Optional keyword argument to pass benchmark predictions.
+        y_true : time series in sktime compatible data container format
+            Ground truth (correct) target values
+            y can be in one of the following formats:
+            Series scitype: pd.Series, pd.DataFrame, or np.ndarray (1D or 2D)
+            Panel scitype: pd.DataFrame with 2-level row MultiIndex,
+                3D np.ndarray, list of Series pd.DataFrame, or nested pd.DataFrame
+            Hierarchical scitype: pd.DataFrame with 3 or more level row MultiIndex
+        y_pred :time series in sktime compatible data container format
+            Forecasted values to evaluate
+            must be of same format as y_true, same indices and columns if indexed
 
         Returns
         -------
-        loss : float
-            Calculated loss metric.
+        loss : float or np.ndarray
```

Comment:
Just wanted to raise that for prob metrics I have float or pd.series (or pd.DataFrame if by eval-by-index). It would be desirable that the outputs of metrics and prob metrics are consistent, so we should decide which format works best.

Reply:
for completeness, can you kindly list the different output types, by condition? As per your mental model?
I might have gotten confused here.

Regarding the case of `evaluate` and `multiindex`, we need to keep this `np.ndarray`, at least for a deprecation cycle, because of downwards compatibility.
Same for the proba metrics - it needs to stay whatever it was, until it underwent a deprecation cycle.

Mid-term, what should it be? I feel we can´t change the numpy output easily, due to interface compatibility with sklearn.